storage-proofs/src/merkle.rs

```diff
@@ -24,11 +24,11 @@ pub type VecMerkleTree<T, A> = merkle::MerkleTree<T, A, VecStore<T>>;
 #[derive(Default, Debug, Clone, Serialize, Deserialize)]
 pub struct MerkleProof<H: Hasher> {
     pub root: H::Domain,
-    path: Vec<(H::Domain, bool)>,
-    leaf: H::Domain,
+    pub path: Vec<(H::Domain, bool)>,
+    pub leaf: H::Domain,
 
     #[serde(skip)]
-    _h: PhantomData<H>,
+    pub _h: PhantomData<H>,
```

Comment:
Instead of making this public, can we have a simple constructor that takes care of this, like we do for other structs with phantom data?